### PR TITLE
Unsubscribing from widgets when page is unloaded

### PR DESCRIPTION
**Pull request by guimafelipe**
## Summary of the pull request
When the Dashboard page is unloaded for every other reason than navigating from it, DevHome keeps subscribed to the widgets. This way, if the DevHome for example is closed while in dashboard, the widgets will keep active and consuming resources.

This adds the un-subscription method  and calls it when the dashboard is getting unloaded, to be more general and consistent.
## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs [hunk 1]
```diff
@@ -656,10 +666,7 @@ protected override void OnNavigatingFrom(NavigatingCancelEventArgs e)
         Log.Logger()?.ReportDebug("DashboardView", $"Leaving Dashboard, deactivating widgets.");
 
         // Deactivate widgets if we're not on the Dashboard.
-        foreach (var widget in PinnedWidgets)
-        {
-            widget.UnsubscribeFromWidgetUpdates();
-        }
+        UnsubscribeFromWidgets();
```
Comment (AmelBawa-msft) [inline tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs:669]: Do we still need this here? Would `Unloaded` handle this case as well?

Comment (guimafelipe) [reply tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs]: It was not needed, but didn't hurt that much to fave. Removed.

**Review by AmelBawa-msft — APPROVED** [review]

File: tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs [hunk 2]
```diff
@@ -119,6 +119,17 @@ private async Task OnLoadedAsync()
     private void OnUnloaded()
     {
         Application.Current.GetService<IAdaptiveCardRenderingService>().RendererUpdated -= HandleRendererUpdated;
+
+        Log.Logger()?.ReportDebug("DashboardView", $"Leaving Dashboard, deactivating widgets.");
+        UnsubscribeFromWidgets();
+    }
+
+    private void UnsubscribeFromWidgets()
```
Comment (dhoehna) [inline tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs:127]: Should this be `async`?  `UnsubscribeFromWidgets` is running on the UI thread, ya?

**Review by dhoehna — APPROVED** [review]

File: tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs (hunk 2 above)
Comment (krschau) [reply tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs:127]: If this weren't in the View but rather the ViewModel, would it maybe not be running on the UI thread and we wouldn't need to Task.Run it? (At some point I want to move a bunch of stuff from View to VM)

**Review by krschau — APPROVED** [review]

**Pull request merged**